mmhuman3d/core/visualization/renderer/torch3d_renderer/meshes.py

```diff
@@ -0,0 +1,540 @@
+                 model_type: str = 'smpl',
+                 N_individual_overdide: int = None,
+                 *,
+                 verts_normals: torch.Tensor = None,
+                 **pose_params) -> None:
+
+        if isinstance(meshes, Meshes):
+            verts = meshes.verts_padded()
+            faces = meshes.faces_padded()
+            textures = meshes.textures
+
+        self.model_type = body_model._get_name().lower(
+        ) if body_model is not None else model_type
+
+        self.model_class = self.model_classes[self.model_type]
+
+        use_list = False
+
+        # formart verts as verts_padded: (N, V, 3)
+        if verts is None:
+            assert body_model is not None
+            verts = body_model(**pose_params)['vertices']
+        elif isinstance(verts, list):
+            verts = list_to_padded(verts)
+            use_list = True
+        # specify number of individuals
+        if N_individual_overdide is not None:
+            verts = verts.view(
+                -1, self.model_class.NUM_VERTS * N_individual_overdide, 3)
+
+        # the information of _N_individual should be revealed in verts's shape
+        self._N_individual = int(verts.shape[-2] // self.model_class.NUM_VERTS)
+
+        assert verts.shape[1] % self.model_class.NUM_VERTS == 0
+        verts = verts.view(-1, self.model_class.NUM_VERTS * self._N_individual,
+                           3)
+        device = verts.device
+        N, V, _ = verts.shape
+
+        # formart faces as faces_padded: (N, F, 3)
+        if isinstance(faces, list):
+            faces = list_to_padded(faces)
+            self.face_individual = faces[0][:self.model_class.NUM_FACES].to(
+                device)
+        elif faces is None:
+            assert body_model is not None
+            self.face_individual = body_model.faces_tensor[None].to(device)
+            faces = self.get_faces_padded(N, self._N_individual)
+        elif isinstance(faces, torch.Tensor):
+            faces = align_input_to_padded(faces, ndim=3, batch_size=N)
+            self.face_individual = faces[:1, :self.model_class.NUM_FACES].to(
+                device)
+        else:
+            raise ValueError(f'Wrong type of faces: {type(faces)}.')
+
+        assert faces.shape == (N,
+                               self.model_class.NUM_FACES * self._N_individual,
+                               3)
+        F = faces.shape[1]
+        if textures is None:
+            if texture_images is None:
+                # input vertex_color should be
+                #   (3), (1, 3), (1, 1, 3). all the same color
+                #   (V, 3), (1, V, 3), each vertex has a single color
+                #   (N, V, 3), each batch each vertex has a single color
+                if isinstance(vertex_color, (tuple, list)):
+                    vertex_color = torch.Tensor(vertex_color)
+                elif isinstance(vertex_color, (torch.Tensor, np.ndarray)):
```

Comment:
`np.ndarray` is the only valid type in this branch. `torch.Tensor` can be removed.

Reply:
ok